src/main/kotlin/org/partiql/scribe/targets/trino/TrinoExpandStruct.kt

```diff
@@ -0,0 +1,336 @@
+package org.partiql.scribe.targets.trino
+
+import org.partiql.plan.Fn
+import org.partiql.plan.Rex
+import org.partiql.plan.rex
+import org.partiql.plan.rexOpCallStatic
+import org.partiql.plan.rexOpCollection
+import org.partiql.plan.rexOpLit
+import org.partiql.plan.rexOpPathKey
+import org.partiql.plan.rexOpStruct
+import org.partiql.plan.rexOpStructField
+import org.partiql.scribe.asNonNullable
+import org.partiql.types.AnyOfType
+import org.partiql.types.BoolType
+import org.partiql.types.CollectionType
+import org.partiql.types.DateType
+import org.partiql.types.DecimalType
+import org.partiql.types.FloatType
+import org.partiql.types.IntType
+import org.partiql.types.ListType
+import org.partiql.types.MissingType
+import org.partiql.types.NullType
+import org.partiql.types.NumberConstraint
+import org.partiql.types.StaticType
+import org.partiql.types.StringType
+import org.partiql.types.StructType
+import org.partiql.types.TimeType
+import org.partiql.types.TimestampType
+import org.partiql.types.function.FunctionParameter
+import org.partiql.types.function.FunctionSignature
+import org.partiql.value.PartiQLTimestampExperimental
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.PartiQLValueType
+import org.partiql.value.stringValue
+import org.partiql.value.symbolValue
+
+private fun StaticType.expand() = this.metas["EXPAND"] == true
+
+internal fun StaticType.toRexTrino(prefixPath: Rex): Rex {
+    if (!this.expand()) {
+        return prefixPath
+    }
```

Comment:
(self-review) Essentially the same as what was deleted from `TrinoRewriter`. Adds this additional check to not expand structs/arrays that do not have any excluded fields.